pkg/odo/cli/component/create.go

```diff
@@ -381,6 +383,11 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 		return errors.New("this directory already contains a devfile, you can't specify devfile via --devfile")
 	}
 
+	// we check if the devfile is already present or not, this location is important - check should happen early
```

Comment:
yeah so we can work on that as part of https://github.com/openshift/odo/issues/4313

Reply:
Of course as a part of that issue. I make this kind of comments on PRs so that on the original issue, GH leaves a reference for us and later we don't have to argue why something makes sense. :wink: